enhancements/olm/phone-home-support.md

```diff
@@ -0,0 +1,145 @@
+# phone-home-support
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ]  -facing documentation is created in [openshift/docs]
+
+## Summary
+
+The purpose of this enhancement is to propose a method of communicating the health of installed operators through [Operator-Lifecycle-Manager (OLM)](https://github.com/operator-framework/operator-lifecycle-manager) to Telemeter.
+
+Today, OLM provides Telemeter with information regarding `subscription syncs` on clusters but fails to communicate the health of installed operators. By communicating which versions of operators are consistently failing, OpenShift engineers and ISV partners could proactively address issues and improve the customer experience.
+
+OLM will communicate the health of installed operators by searching for CatalogSourceVersions (CSV)s and reporting their name, phase, version, and reason fields.
+
+The initial target for this enhancement is to support reporting installed operator health per version to Telemeter in OpenShift 4.2 and 4.3.
+
+## References
+
+- https://docs.google.com/document/d/16EOP2W97EEIfkQ0FAwz3btwtMmKNDBM0qXrnDhXfzsM/edit
+- https://github.com/operator-framework/operator-lifecycle-manager/tree/master/pkg/metrics
+
+## Motivation
+
+The primary motivation for this enhancement is to enable OLM to collect information about the health per version of CSVs installed on the cluster and report these metrics to Telemeter. With this information, key insights regarding operator health could be delivered to both OpenShift Engineering as well as ISVs.
+
+### Goals
+
+- Reporting CSV health per version
+- Backport reporting CSV health per version to 4.2
+
+### Non-Goals
+
+- Reporting CSV Status Conditions per version
+- Integrating with Insights Operator in 4.3
+
+## Proposal
+
+### User Stories
```

Comment:
Do we plan on creating an alert for failing CSV's ? 

Is there a way to only do this for CSV's that are from a catalog source we care about? Does that restriction make sense?

Reply:
>Do we plan on creating an alert for failing CSV's ?

Yes - I will update the PR to include this information.

>Is there a way to only do this for CSV's that are from a catalog source we care about? Does that restriction make sense?

ClusterAdmins (not using Telemeter) may be interested in collecting said metrics about CSVs from `CatalogSource`s that they define. The Telemeter alerts should be limited to the `CatalogSource`s that we care about.

Reply:
@shawn-hurley this comment has been addressed in the latest commit.

Reply:
👍